app/models/user.rb

```diff
@@ -61,8 +61,12 @@ def pciu_alternate_phone
     pciu&.get_alternate_phone&.to_s
   end
 
-  def first_name
-    identity.first_name || (mhv_icn.present? ? mpi&.profile&.given_names&.first : nil)
+  def first_name(args = { mpi: false })
```

Comment:
consider replacing this with the more idiomatic: [ruby keyword arguments](https://thoughtbot.com/upcase/videos/ruby-keyword-arguments)

Or if you don't want named arguments consider using splat syntax (though that is less clear / readable).

Additionally, since you're running into some complexity issues, I wonder if it would make sense to use the decorator patten (while admittedly not a fan, it might be a good way to abstract the source / origin of some these first names.